### ISSUE TITLE
Question: How to restore to latest (backup + WAL file from the archive)

### ISSUE THREAD

**Issue by AriAriOctopus** [pgbackrest/pgbackrest#1974]
Please provide the following information when submitting an issue (feature requests or general comments can skip this):

1. pgBackRest version:
pgBackRest 2.43

2. PostgreSQL version:
postgres (PostgreSQL) 13

3. Operating system/version - if you have more than one server (for example, a database server, a repository host server, one or more standbys), please specify each:
CentOS7

4. Did you install pgBackRest from source or from a package?
package - yum
pgbackrest.x86_64                 2.43-1.rhel7                 @pgdg-common

5. Please attach the following as applicable:
[patroni-show-config.txt](https://github.com/pgbackrest/pgbackrest/files/10387273/patroni-show-config.txt)
[pgbackrest.conf.txt](https://github.com/pgbackrest/pgbackrest/files/10387203/pgbackrest.conf.txt)
[postgresql.conf.txt](https://github.com/pgbackrest/pgbackrest/files/10387204/postgresql.conf.txt)
[error_logs.txt](https://github.com/pgbackrest/pgbackrest/files/10387297/error_logs.txt)

7. Describe the issue:
I have a 3 node patroni cluster, 2 of the replicas are out of sync with the leader and need to be reinitialized.
I am unable to restore the database such that replication can resume because the WAL files are always missing from the leader.
I keep 1TB of WAL but that is insufficient given the backup strategy.
To re-join a node I have to take multiple incremental backups until the DB is within the wal_keep_size window, this is error-prone and tedious. I have tried various configuration options and commands but can't seem to solve this one.
Is there a way to run a restore such that:
It uses the latest backup set and then reads in all the WAL files from the S3 archive?
The latest WAL file from the archive is re-calculated once the initial restore is complete, because by the time the restore has completed a lot more WAL has been upload to S3?
All of this can work with patroni rather than pgbackrest (appreciate this might be out of scope here)
Also if anyone has a better solution, I'd love to hear it because I don't think it should be this hard :)

**Comment by m-borger** [comment]
I have no knowledge of patroni, don't know if this has specific behaviour, but just curious about this: Normally, when you would restore and recover a database cluster, ```restore_command``` would be set to some  ```archive-get``` command. This would enable that cluster to restore the archives needed from the repository (instead of getting them from the wal stream, limited to the ```wal_keep_size``` ).
Only if you would backup so often that those archives are already expired, you would get an error, however different from the one in your logs. (Somehting like ```unable to find 000000xx000aaaaa000000bb in the archive``` instead of ```FATAL: could not receive data from WAL stream:```) . Or am I missing something?

**Comment by AriAriOctopus** [comment]
Thanks for the feedback. I haven't had a chance to test these suggestions out yet but when I do I will post my results here. The create replace method is something I also tried but without the recovery conf.
